tests/autoyast/verify_ext4.pm

```diff
@@ -0,0 +1,61 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate autoyast_ext4 profile.
+#          * Verify registration
+#          * Verify partitioning: ext4 and swap
+#          * Verify software packages/patterns
+#          * Verify users
+# Maintainer: Joaquín Rivera <jeriveramoya@suse.com>
+
+use base 'basetest';
+use strict;
+use testapi;
+
+sub verify_registration {
+    my $output = script_output 'SUSEConnect --list-extensions';
+    unless ($output =~ /Basesystem.*Activated/ && $output =~ /Applications Server.*Activated/) {
+        die 'Registered system does not contains Basesystem and/or Applications Server';
+    }
+}
+
+sub verify_partitioning {
+    assert_script_run 'blkid /dev/vda | grep PTTYPE="dos"';
+    assert_script_run 'grep "swap\s\+swap" /etc/fstab';
+    assert_script_run 'grep "/\s\+ext4" /etc/fstab';
+}
+
+sub verify_software {
+    my @packages = qw(glibc grub2 snapper syslinux kdump sles-release kexec-tools
+      patterns-base-apparmor patterns-base-base patterns-base-basesystem
+      patterns-base-documentation patterns-base-enhanced_base
+      patterns-base-minimal_base patterns-base-x11
+      patterns-base-x11_enhanced patterns-gnome-gnome_basic
+      patterns-gnomegnome_basis);
+    assert_script_run 'rpm -q ' . join(" ", @packages);
+}
+
+sub verify_users {
+    my $output = script_output 'cat /etc/passwd';
+    my $root   = 'root:x:0:0:root:/root:/bin/bash';
+    my $user   = 'bernhard:x:1000:100:Bernhard M. Wiedemann:/home/bernhard:/bash';
+    unless ($output =~ /$root/ && $output =~ /$user/) {
+        die 'Expected users in the system provide wrong configuration in /etc/passwd';
+    }
+}
+
+sub run {
+    select_console 'root-console';
+    verify registration;
```

Comment:
First of all, there is a typo, secondly, this makes sense only on SLE 15, on SLE 12 we have unregistered system.